src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/RazorSyntaxFacts.cs

```diff
@@ -134,4 +137,54 @@ public static bool IsAnyEndTag(RazorSyntaxNode n)
 
     public static bool IsInCodeBlock(RazorSyntaxNode n)
         => n.FirstAncestorOrSelf<RazorSyntaxNode>(n => n is RazorDirectiveSyntax { DirectiveDescriptor.Directive: "code" }) is not null;
+
+    internal static bool TryGetNamespaceFromDirective(RazorDirectiveSyntax directiveNode, [NotNullWhen(true)] out string? @namespace)
+    {
+        foreach (var child in directiveNode.DescendantNodes())
+        {
+            if (child.GetChunkGenerator() is AddImportChunkGenerator { IsStatic: false } usingStatement)
+            {
+                @namespace = usingStatement.ParsedNamespace;
+                return true;
+            }
+        }
+
+        @namespace = null;
+        return false;
+    }
+
+    internal static bool IsInUsingDirective(RazorSyntaxNode node)
+    {
+        var directives = node
+            .AncestorsAndSelf()
+            .OfType<RazorDirectiveSyntax>();
+
+        foreach (var directive in directives)
+        {
+            if (TryGetNamespaceFromDirective(directive, out var _))
```

Comment:
If this is meaningly different from `directive.IsUsingDirective`, as per my above comment, then should probably rename the method to `IsInSimpleUsingDirective` or something

Reply:
I changed to use `IsUsingDirective` for now. I think what will end up happening is even it will gracefully fail if `TryGetNamespaceFromDirective` but I need to add tests here. Coming up in a future change.